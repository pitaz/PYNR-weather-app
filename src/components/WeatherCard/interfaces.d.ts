export interface WeatherCardProps {
  data: any;
  unit?: string;
  setTime: any;
  setTemp: any;
}

export interface Main {
  temp: string
}
export interface Items {
  main: Main;
  dt_txt: string;
  weather: any;
}