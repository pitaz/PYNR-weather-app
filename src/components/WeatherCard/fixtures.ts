export const data = [
  "27",
  [
      {
          "dt": 1630022400,
          "main": {
              "temp": 11.26,
              "feels_like": 10.82,
              "temp_min": 10.37,
              "temp_max": 11.26,
              "pressure": 1015,
              "sea_level": 1015,
              "grnd_level": 953,
              "humidity": 91,
              "temp_kf": 0.89
          },
          "weather": [
              {
                  "id": 803,
                  "main": "Clouds",
                  "description": "broken clouds",
                  "icon": "04n"
              }
          ],
          "clouds": {
              "all": 68
          },
          "wind": {
              "speed": 3.35,
              "deg": 255,
              "gust": 8.98
          },
          "visibility": 10000,
          "pop": 0.64,
          "sys": {
              "pod": "n"
          },
          "dt_txt": "2021-08-27 00:00:00"
      },
      {
          "dt": 1630033200,
          "main": {
              "temp": 10.25,
              "feels_like": 9.76,
              "temp_min": 9.52,
              "temp_max": 10.25,
              "pressure": 1014,
              "sea_level": 1014,
              "grnd_level": 953,
              "humidity": 93,
              "temp_kf": 0.73
          },
          "weather": [
              {
                  "id": 500,
                  "main": "Rain",
                  "description": "light rain",
                  "icon": "10n"
              }
          ],
          "clouds": {
              "all": 87
          },
          "wind": {
              "speed": 2.46,
              "deg": 249,
              "gust": 6.7
          },
          "visibility": 10000,
          "pop": 0.53,
          "rain": {
              "3h": 0.2
          },
          "sys": {
              "pod": "n"
          },
          "dt_txt": "2021-08-27 03:00:00"
      },
      {
          "dt": 1630044000,
          "main": {
              "temp": 9.67,
              "feels_like": 7.84,
              "temp_min": 9.67,
              "temp_max": 9.67,
              "pressure": 1015,
              "sea_level": 1015,
              "grnd_level": 953,
              "humidity": 97,
              "temp_kf": 0
          },
          "weather": [
              {
                  "id": 500,
                  "main": "Rain",
                  "description": "light rain",
                  "icon": "10d"
              }
          ],
          "clouds": {
              "all": 99
          },
          "wind": {
              "speed": 3.5,
              "deg": 266,
              "gust": 8.98
          },
          "visibility": 10000,
          "pop": 0.88,
          "rain": {
              "3h": 0.94
          },
          "sys": {
              "pod": "d"
          },
          "dt_txt": "2021-08-27 06:00:00"
      },
      {
          "dt": 1630054800,
          "main": {
              "temp": 10.27,
              "feels_like": 9.86,
              "temp_min": 10.27,
              "temp_max": 10.27,
              "pressure": 1016,
              "sea_level": 1016,
              "grnd_level": 954,
              "humidity": 96,
              "temp_kf": 0
          },
          "weather": [
              {
                  "id": 500,
                  "main": "Rain",
                  "description": "light rain",
                  "icon": "10d"
              }
          ],
          "clouds": {
              "all": 100
          },
          "wind": {
              "speed": 3.86,
              "deg": 264,
              "gust": 8.29
          },
          "visibility": 10000,
          "pop": 0.98,
          "rain": {
              "3h": 1.02
          },
          "sys": {
              "pod": "d"
          },
          "dt_txt": "2021-08-27 09:00:00"
      },
      {
          "dt": 1630065600,
          "main": {
              "temp": 12.16,
              "feels_like": 11.73,
              "temp_min": 12.16,
              "temp_max": 12.16,
              "pressure": 1016,
              "sea_level": 1016,
              "grnd_level": 955,
              "humidity": 88,
              "temp_kf": 0
          },
          "weather": [
              {
                  "id": 500,
                  "main": "Rain",
                  "description": "light rain",
                  "icon": "10d"
              }
          ],
          "clouds": {
              "all": 96
          },
          "wind": {
              "speed": 3.68,
              "deg": 267,
              "gust": 6.72
          },
          "visibility": 10000,
          "pop": 1,
          "rain": {
              "3h": 0.62
          },
          "sys": {
              "pod": "d"
          },
          "dt_txt": "2021-08-27 12:00:00"
      },
      {
          "dt": 1630076400,
          "main": {
              "temp": 14.96,
              "feels_like": 14.26,
              "temp_min": 14.96,
              "temp_max": 14.96,
              "pressure": 1015,
              "sea_level": 1015,
              "grnd_level": 954,
              "humidity": 67,
              "temp_kf": 0
          },
          "weather": [
              {
                  "id": 500,
                  "main": "Rain",
                  "description": "light rain",
                  "icon": "10d"
              }
          ],
          "clouds": {
              "all": 69
          },
          "wind": {
              "speed": 3.78,
              "deg": 294,
              "gust": 5.86
          },
          "visibility": 10000,
          "pop": 0.85,
          "rain": {
              "3h": 0.21
          },
          "sys": {
              "pod": "d"
          },
          "dt_txt": "2021-08-27 15:00:00"
      },
      {
          "dt": 1630087200,
          "main": {
              "temp": 11.13,
              "feels_like": 10.57,
              "temp_min": 11.13,
              "temp_max": 11.13,
              "pressure": 1016,
              "sea_level": 1016,
              "grnd_level": 955,
              "humidity": 87,
              "temp_kf": 0
          },
          "weather": [
              {
                  "id": 500,
                  "main": "Rain",
                  "description": "light rain",
                  "icon": "10d"
              }
          ],
          "clouds": {
              "all": 70
          },
          "wind": {
              "speed": 1.84,
              "deg": 267,
              "gust": 2.4
          },
          "visibility": 10000,
          "pop": 0.77,
          "rain": {
              "3h": 0.22
          },
          "sys": {
              "pod": "d"
          },
          "dt_txt": "2021-08-27 18:00:00"
      },
      {
          "dt": 1630098000,
          "main": {
              "temp": 9.48,
              "feels_like": 8.37,
              "temp_min": 9.48,
              "temp_max": 9.48,
              "pressure": 1017,
              "sea_level": 1017,
              "grnd_level": 955,
              "humidity": 94,
              "temp_kf": 0
          },
          "weather": [
              {
                  "id": 500,
                  "main": "Rain",
                  "description": "light rain",
                  "icon": "10n"
              }
          ],
          "clouds": {
              "all": 48
          },
          "wind": {
              "speed": 2.28,
              "deg": 241,
              "gust": 3.84
          },
          "visibility": 10000,
          "pop": 0.29,
          "rain": {
              "3h": 0.12
          },
          "sys": {
              "pod": "n"
          },
          "dt_txt": "2021-08-27 21:00:00"
      }
  ]
]