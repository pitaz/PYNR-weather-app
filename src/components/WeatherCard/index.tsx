import { Card, CardContent, makeStyles, Typography } from "@material-ui/core";
import React, { FC } from "react";

export interface WeatherCardProps {
  data: []
}

const WeatherCard: FC<WeatherCardProps> = ({data}) => {
  const classes = useStyles();
  console.log('data :>> ', data[1]);
  return (
    <Card className={classes.card}>
      <CardContent>
        <Typography
          className={classes.title}
          color="textSecondary"
          gutterBottom
        >
         Temperature
        </Typography>
        <Typography
          className={classes.temperature}
          color="textSecondary"
          gutterBottom
        >
         5 C
        </Typography>
        <Typography
          className={classes.date}
          color="textSecondary"
          gutterBottom
        >
         15 Sept, 2021
        </Typography>
      </CardContent>
    </Card>
  );
};

const useStyles = makeStyles((theme) => ({
  card: {
    height: 240,
    width: 204,
  },
  bar: {
    padding: "20px 30px",
  },
  title: {
    fontSize: 14,
  },
  temperature: {
    fontSize: 64,
  },
  date: {
    fontSize: 20,
  }
}));

export default WeatherCard;
