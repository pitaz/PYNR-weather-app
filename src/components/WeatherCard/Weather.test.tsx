/**
 * @jest-environment jsdom
 */

 import React from "react";
 import { render, screen } from "@testing-library/react";

import WeatherCard from ".";
import {data} from './fixtures';

describe('Weather card Component', () => {
  const props = { 
    data,
    setTime: '',
    setTemp: ''
   };
  const wrapper = render(<WeatherCard {...props} />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
