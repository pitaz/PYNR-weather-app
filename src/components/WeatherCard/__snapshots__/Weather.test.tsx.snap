// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather card Component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <p
            class="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Temperature
          </p>
          <p
            class="MuiTypography-root makeStyles-temperature-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            11.26
             
            F
          </p>
          <p
            class="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Clouds
          </p>
          <p
            class="MuiTypography-root makeStyles-date-5 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            27 Aug 2021
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Temperature
        </p>
        <p
          class="MuiTypography-root makeStyles-temperature-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          11.26
           
          F
        </p>
        <p
          class="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Clouds
        </p>
        <p
          class="MuiTypography-root makeStyles-date-5 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          27 Aug 2021
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
